### PR TITLE
feat(mac): both option keys generate right alt if no left alt mapping

### PR DESCRIPTION
**Pull request by sgschantz**
When a keyman keyboard does not define any rules for the Left Alt key, then both the left and right option keys on the Mac keyboard should result in a Right Alt key combination being generated. 

Fixes: #875
Fixes: #933

# User Testing

* **TEST_RALT_KEYBOARD_WITH_BOTH_OPTION_KEYS**: Khmer Angkor uses only RALT rules

1. Install this version of Keyman and switch to the Khmer Angkor keyboard
1. Press <kbd>left-option-Z</kbd>
1. Confirm that this generates the character '<'
1. Press <kbd>right-option-Z</kbd>
1. Confirm that this generates the character '<'

* **TEST_RALT_KEYBOARD_WITH_OPTION_KEY_COMBOS**: Armenian Mnemonic uses RALT with SHIFT

1. Install and switch to the Armenian Mnemonic keyboard
1. Press <kbd>left-option-9</kbd >
1. Confirm that this generates the subscript character '₉'
1. Press <kbd>shift-left-option-9</kbd >
1. Confirm that this generates the superscript character '⁹'
1. Press <kbd>shift-right-option-9</kbd >
1. Confirm that this generates the same superscript character '⁹'

* **TEST_LALT_KEYBOARD_WITH_BOTH_OPTION_KEYS**: Devanagari Typewriter uses only LALT rules

1. Install and switch to the Devanagari Typewriter (SIL) keyboard
1. Press <kbd> control-left-option-COMMA</kbd>
1. Confirm that this generates the DEVANAGARI DOUBLE DANDA character '॥'
1. Press <kbd> control-right-option-COMMA </kbd>
1. Confirm that this generates the comma character ','

### CODE REVIEW

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m [hunk 1]
```diff
@@ -67,6 +64,38 @@ -(void)changeKeyboardWithKmxFilePath:(NSString*) path {
   }
 }
 
+-(CoreKeyboardInfo*) getKeyboardInfoForKmxFile:(NSString*)kmxFile {
+  NSArray *keyArray = [self getKeyArray];
+  CoreKeyboardInfo* info = [[CoreKeyboardInfo alloc] init:kmxFile keyArray: keyArray];
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m:69]: Is keyArray and its members leaked here?

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m [hunk 2]
```diff
@@ -67,6 +64,38 @@ -(void)changeKeyboardWithKmxFilePath:(NSString*) path {
...
+}
+
+/**
+ * Get the list of Keys supported by the keyboard
+ */
+-(NSArray*)getKeyArray {
+  NSMutableArray *keyArray = [[NSMutableArray alloc] initWithCapacity:0];
+  km_core_keyboard_key *keyList;
+
+  km_core_status result = km_core_keyboard_get_key_list(self.coreKeyboard, &keyList);
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m:80]: Is keyList leaked here?

**Review by mcdurdin — APPROVED** [review]
This LGTM, aside from concerns on memory management. Good work

**Comment by dinakaranr** [comment]
# Test Results
I tested this issue with the attached "Keyman 18.0.118-alpha-local" build on the macOS Sonoma 14.5 version. here, I am sharing my observation.

* **TEST_RALT_KEYBOARD_WITH_BOTH_OPTION_KEYS (passed):** 
1. Installed the "Keyman 18.0.118-alpha-local" build.
2. Installed the Khmer Angkor keyboard
3. Select the Khmer Angkor keyboard
4. Press left-option-Z
5. Verified that this generates the character '<'
6. Press right-option-Z
7. Verified that this generates the character '<'

* **TEST_RALT_KEYBOARD_WITH_OPTION_KEY_COMBOS (passed):** 
1. Installed the "Keyman 18.0.118-alpha-local" build.
2. Installed the "Armenian Mnemonic" keyboard
3. Select the Armenian Mnemonic keyboard
4. Press left-option-9
5. Verified that this generates the subscript character '₉'
6. Press shift-left-option-9
7. Verified that this generates the superscript character '⁹'
8. Press shift-right-option-9
9. Verified that this generates the same superscript character '⁹'

* **TEST_LALT_KEYBOARD_WITH_BOTH_OPTION_KEYS (passed):** 
1. Installed the "Keyman 18.0.118-alpha-local" build.
2. Installed the "Devanagari Typewrite" keyboard
3. Select the Devanagari Typewriter (SIL) keyboard
4. Press control-left-option-COMMA
5. Verified that this generates the DEVANAGARI DOUBLE DANDA character '॥'
6. Press control-right-option-COMMA
7. Verified that this generates the comma character ','

It works well. Thank you.

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m (hunk 2 above)
Comment (sgschantz) [reply mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m:80]: yes, adding a call to km_core_keyboard_key_list_dispose

File: mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m (hunk 1 above)
Comment (sgschantz) [reply mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m:69]: no, by default there is a strong reference to the key array in CoreKeyboardInfo which is dealloced when the keyboard is changed

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.122-alpha](https://keyman.com/downloads/releases/alpha/18.0.122)

**Pull request merged**